cli/types/cypress.d.ts

```diff
@@ -130,6 +130,53 @@ declare namespace Cypress {
     unsupportedVersion?: boolean
   }
 
+  interface Ensure {
+    /**
+     * Throws an error if `subject` is not one of the passed in `type`s.
+     */
+    isType(subject: any, type: PrevSubject[], commandName: string, cy: $Cy): void
+
+    /**
+     * Throws an error if `subject` is not a DOM element.
+     */
+    isElement(subject: any, commandName: string, cy: Chainable): void
+
+    /**
+     * Throws an error if `subject` is not a `window`.
+     */
+    isWindow(subject: any, commandName: string, cy: Chainable): void
+
+    /**
+     * Throws an error if `subject` is not a `document`.
+     */
+    isDocument(subject: any, commandName: string, cy: Chainable): void
+
+    /**
+     * Throws an error if `subject` is not a DOM element attached to the application under test.
+     */
+    isAttached(subject: any, commandName: string, cy: $Cy, onFail?: any): void
```

Comment:
Should `onFail()` be a function?

Reply:
Changed the type to `Log | Function`. It's possible that Function could be made stricter, but I'd like to leave that aside in this PR, until we decide to more strictly type the internals that this is pointing towards.